bin/check-influxdb-metrics.rb

```diff
@@ -79,32 +79,53 @@ class CheckInfluxDbMetrics < Sensu::Plugin::Check::CLI
          long: '--filter=VALUE',
          description: 'Set the name of your filter, for example: `datacenter`'
 
-  def encode_parameters(parameters)
-    encodedparams = Addressable::URI.escape(parameters)
-    "#{config[:db]}&q=" + encodedparams
-  end
+  option :period,
+         long: '--period=VALUE',
+         description: 'Filter by a given day instead of the last 10 minutes'
 
   def filter_by_environment_when_needed
     config[:tag].nil? && config[:filter].nil? ? '' : " AND \"#{config[:tag]}\" =~ /#{config[:filter]}/"
   end
 
-  def yesterday_query # Reads the value from 10 minutes before yesterday at this time.
-    query = "SELECT sum(\"value\") from \"#{config[:metric]}\" WHERE time > now() - 1455m AND time < now() - 1445m"
+  def base_query
+    "SELECT sum(\"value\") from \"#{config[:metric]}\" "
+  end
+
+  def period
+    config[:period].nil? ? 10 : config[:period].to_i # by default set a period of 10 minutes
```

Comment:
you can set the default in the option parser (much cleaner), see here for an example: https://github.com/pliyosenpai/sensu-plugins-influxdb-metrics-checker/blob/328ca9dc21c2950c56e0ab1820d1120621ce7897/bin/check-influxdb-metrics.rb#L56-L61